src/OrbitGl/TrackContainer.h

```diff
@@ -0,0 +1,129 @@
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_TRACK_CONTAINER_H_
+#define ORBIT_GL_TRACK_CONTAINER_H_
+
+#include <ClientData/CaptureData.h>
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "CaptureViewElement.h"
+#include "ClientData/CaptureData.h"
+#include "ClientData/TimerChain.h"
+#include "TimeGraphLayout.h"
+#include "Track.h"
+#include "TrackManager.h"
+#include "Viewport.h"
+#include "absl/container/flat_hash_map.h"
+
+namespace orbit_gl {
+
+// Represent the space where Tracks will be drawn.
+class TrackContainer final : public CaptureViewElement {
+ public:
+  explicit TrackContainer(CaptureViewElement* parent, TimelineInfoInterface* timeline_info,
+                          Viewport* viewport, TimeGraphLayout* layout, OrbitApp* app,
+                          orbit_client_data::CaptureData* capture_data);
+
+  [[nodiscard]] float GetHeight() const override;
+
+  [[nodiscard]] const orbit_client_data::CaptureData* GetCaptureData() const {
+    return capture_data_;
+  }
+  [[nodiscard]] TrackManager* GetTrackManager() { return track_manager_.get(); }
+  [[nodiscard]] orbit_client_data::ThreadTrackDataProvider* GetCaptureDataProvider() {
+    return thread_track_data_provider_;
+  }
+
+  [[nodiscard]] float GetTextBoxHeight() const { return layout_.GetTextBoxHeight(); }
+
+  void VerticalZoom(float real_ratio, float mouse_screen_y_position);
+  void VerticallyMoveIntoView(const orbit_client_protos::TimerInfo& timer_info);
+  void VerticallyMoveIntoView(Track& track);
+
+  [[nodiscard]] const orbit_client_protos::TimerInfo* FindPreviousFunctionCall(
+      uint64_t function_address, uint64_t current_time,
+      std::optional<uint32_t> thread_id = std::nullopt) const;
+  [[nodiscard]] const orbit_client_protos::TimerInfo* FindNextFunctionCall(
+      uint64_t function_address, uint64_t current_time,
+      std::optional<uint32_t> thread_id = std::nullopt) const;
+  [[nodiscard]] std::vector<const orbit_client_protos::TimerInfo*> GetAllTimersForHookedFunction(
+      uint64_t function_address) const;
+
+  void SetThreadFilter(const std::string& filter);
+
+  [[nodiscard]] std::vector<const orbit_client_data::TimerChain*> GetAllThreadTrackTimerChains()
+      const;
+  [[nodiscard]] int GetNumVisiblePrimitives() const;
+
+  [[nodiscard]] const TimeGraphLayout& GetLayout() const { return layout_; }
+  [[nodiscard]] TimeGraphLayout& GetLayout() { return layout_; }
+
+  [[nodiscard]] const orbit_client_protos::TimerInfo* FindPrevious(
+      const orbit_client_protos::TimerInfo& from);
+  [[nodiscard]] const orbit_client_protos::TimerInfo* FindNext(
+      const orbit_client_protos::TimerInfo& from);
+  [[nodiscard]] const orbit_client_protos::TimerInfo* FindTop(
+      const orbit_client_protos::TimerInfo& from);
+  [[nodiscard]] const orbit_client_protos::TimerInfo* FindDown(
+      const orbit_client_protos::TimerInfo& from);
```

Comment:
WDYT, should these 4 functions maybe be in `TrackManager` instead? I'm ok with keeping them here, but they also don't fully belong here IMO.

Reply:
CaptureDataProvider should take care in the future. After our discussion I will move back to TimeGraph and reconsider this later. 

But let me know if I understand correctly what part to move to TrackManager.

This function is being called by JumpToNeighborTimer (currently TimeGraph) who after finding is making a SelectAndMakeVisible (currently in TimeGraph, consists in HorizontalMove(TimelineInfoInterface) + VerticalMove(TrackContainer) + SelectTimer()(App))

Your proposal here is only moving temporarily Find functions to TrackManager and nothing else? In that case I would  rather do the effort moving them directly to the provider.

Reply:
I think the `FindPrevious/NextFunctionCall` methods make sense to move to `CaptureDataProvider` (and for now keep them in `TimeGraph`), because they are conceptually not related to any visual representation.

My comment here is just referring to the 4 methods above (`FindPrevious`, `FindNext`, `FindTop`, `FindDown`). I was under the impression 
they were a good fit for track manager because they require knowledge of the tracks (at least currently, because they call `GetOrCreateTrack`) and update the visual state (scrolling) - but actually, now that I write this, maybe they should actually stay in the `TrackContainer` for exactly this reason...

Reply:
Offline: We decided to maintain these functions here, they are moving from a timer to a closer one.